feat: Разработали приложение "Калькулятор счета"

timachoo opened the pull request:
Разработали приложение "Калькулятор счета"

Commit:
Remove classes

Commit:
Change project name

timachoo committed:
feat: Разработали приложение "Калькулятор счета"

timachoo committed:
Merge branch 'main' into master

gusar-off commented on src/main/java/Messages.java:
```diff
@@ -0,0 +1,11 @@
+public class Messages {
+    public static final String finishPhrase = "Завершить";
```
Константы лучше в SCREAMING_CASE

gusar-off commented on src/main/java/Messages.java:
```diff
@@ -0,0 +1,11 @@
+    public static final String finishPhrase = "Завершить";
+
+    public static boolean Bye(String text){
+        if (text.equalsIgnoreCase(finishPhrase.trim())) {
```
Может лучше наоборот? У константы не будет лишних пробелов
`FINISH_PHASE.equalsIgnoreCase(text.trim())`

gusar-off commented on src/main/java/Bill.java:
```diff
@@ -0,0 +1,35 @@
+    public void printBill(){
+        System.out.println("Добавленные товары:");
+
+        for(int i = 0; i < this.products.size(); i++){
```
Тут можно заменить на цикл foreach
`for (Product product : products) {`

gusar-off commented on src/main/java/Money.java:
```diff
@@ -0,0 +1,25 @@
+
+        String result;
+
+        if (price%100>=11&&price%100<=14){
```
Здесь лучше сначала округлить до `int` с помощью `Math.floor` и это значение уже проверять на остаток от деления.

Eще, кстати, пробелы куда-то потерялись, лучше так:
`if (price % 100 >= 11 && price % 100 <= 14) {`

gusar-off commented on src/main/java/Product.java:
```diff
@@ -0,0 +1,62 @@
+
+public class Product {
+    private static final Scanner scanner = new Scanner(System.in);
+    private String name;
```
Поля, проинициализированные в конструкторе, можно сделать `final`

gusar-off commented on src/main/java/Product.java:
```diff
@@ -0,0 +1,62 @@
+public class Product {
+    private static final Scanner scanner = new Scanner(System.in);
+    private String name;
+    private double price = 0;
```
Числовые примитивы по умолчанию инициализируются 0, соответственно можно не указывать

timachoo committed:
fix: наименование константы finishPhrase исправлено в SCREAMING_CASE
В Messages.Bye исправлена ошибка в сравнении FINISH_PHRASE и вхоящего парвметра, trim должен быть у входящего параметра.

timachoo committed:
fix: в Bill.printBill цикл переписан через foreach

timachoo committed:
fix: приводим параметр price к int, потом работаем с ним для полкчения остатка от деления
форматируем текст, добавляем потерянные пробелы

timachoo committed:
fix: исправлены замечания по полям, price не надо инициализировать в 0 по умолчанию (он и так 0), поля сделаны final

gusar-off commented on src/main/java/Money.java:
```diff
@@ -0,0 +1,26 @@
+    public static String getRurPrice(double price){
+
+        String result;
+        int x = (int)Math.floor(price);
```
Лучше давать более осмысленные названия для переменных. Например, тут можно сделать `roundedPrice`

timachoo committed:
fix: исправлены наименование переменной на осмысленное
